book-2-mvc/chapters/VIEW_MODELS.md

```diff
@@ -0,0 +1,458 @@
+              <span>@dog.Name</span>
+            </div>
+            <div>
+              <label class="font-weight-bold">Breed:</label>
+              <span>@dog.Breed</span>
+            </div>
+            <div>
+              <label class="font-weight-bold">Notes:</label>
+              <p>@dog.Notes</p>
+            </div>
+          </div>
+        </div>
+        }
+      </div>
+    </section>
+
+    <section class="col-lg-4 col-md-8 container mt-5">
+      <h1>Walkers Near Me</h1>
+
+      <ul class="list-group mt-4">
+        @foreach (Walker walker in Model.Walkers) {
+        <li class="list-group-item disabled" aria-disabled="true">
+          <img src="@walker.ImageUrl" style="width:50px" />
+          <span class="font-weight-bold ml-4">@walker.Name</span>
+        </li>
+        }
+      </ul>
+    </section>
+  </div>
+</div>
+```
+
+## Using View Models with Forms
+
+Currently the Create and Edit forms for Owners have a text input field to collect an owner's neighborhood Id. It was mentioned ealier that we'd ideally like to have that be a dropdown list instead. We can make this happen with view models. Once again, lets think about what we'd need to have in _state_ if this were a React application. 
+
+- properties for all the Owner form fields
+- a list of available options for the dropdown
+
+Create a new class inside the ViewModels folder and name it `OwnerFormViewModel`. Add the following code
+
+```csharp
+using System.Collections.Generic;
+
+namespace DogWalker.Models.ViewModels
+{
+    public class OwnerFormViewModel
+    {
+        public Owner Owner { get; set; }
+        public List<Neighborhood> Neighborhoods { get; set; }
+    }
+}
+```
+
+Now inside `OwnersController` add the following helper method that will get us a list of all the neighborhoods in the database.
+
+```csharp
+private List<Neighborhood> GetNeighborhoods()
+{
+    using (SqlConnection conn = Connection)
+    {
+        conn.Open();
+        using (SqlCommand cmd = conn.CreateCommand())
+        {
+            cmd.CommandText = @"SELECT Id, Name FROM Neighborhood";
+
+            SqlDataReader reader = cmd.ExecuteReader();
+
+            List<Neighborhood> neighborhoods = new List<Neighborhood>();
+
+            while (reader.Read())
+            {
+                Neighborhood neighborhood = new Neighborhood()
+                {
+                    Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                    Name = reader.GetString(reader.GetOrdinal("Name"))
+                };
+                neighborhoods.Add(neighborhood);
+            }
+
+            reader.Close();
+
+            return neighborhoods;
+        }
+
+    }
+}
+```
+
+Update the GET `Create` method to now create a view model and pass it to the view
+
+```csharp
+// GET: Owners/Create
+public ActionResult Create()
+{
+    OwnerFormViewModel vm = new OwnerFormViewModel();
+
+    vm.Neighborhoods = GetNeighborhoods();
+    vm.Owner = new Owner();
```

Comment:
This is unnecessary, and I think it may spark questions that we don't really need to answer.